main.go

```diff
@@ -140,55 +150,132 @@ func (d *MDNSDiscovery) StartSync(eventCB discovery.EventCallback, errorCB disco
 	ctx, cancel := context.WithCancel(context.Background())
 	go func() {
 		defer close(queriesChan)
+		queryLoop(ctx, queriesChan)
+	}()
```

Comment:
On the other hand, we keep `close` in the same 'scroll region' as `make`? Just a matter of making it apparent chan is closed when loop func finishes, otherwise we need to look there.

Reply:
OK